src/lib/y2storage/proposal_settings.rb

```diff
@@ -53,11 +99,43 @@ def initialize
       @use_separate_home        = true
       @home_filesystem_type     = Filesystems::Type::XFS
       @enlarge_swap_for_suspend = false
+      @resize_windows           = true
+      @windows_delete_mode      = :ondemand
+      @linux_delete_mode        = :ondemand
+      @other_delete_mode        = :ondemand
     end
 
     def use_encryption
       !encryption_password.nil?
     end
+
+    # Whether the settings disable deletion of a given type of partitions
+    #
+    # @see #windows_delete_mode
+    # @see #linux_delete_mode
+    # @see #other_delete_mode
+    #
+    # @param type [#to_s] :linux, :windows or :other
+    # @return [Boolean]
+    def delete_forbidden(type)
+      send(:"#{type}_delete_mode").to_sym == :none
```

Comment:
OK with public send.
But regarding to_sym, if we want to support to accept both "none" and `:none`, then still I think that internally it should be consistent. so modify `{type}_delete_mode` to accept both, but storage only one. If we declare that method return `Symbol` we should rely on it as also other users of API will rely on it.